download/download.go

```diff
@@ -0,0 +1,87 @@
+package download
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	"log"
+	"net/http"
+	"runtime"
+)
+
+var (
+	_downloadURL = "https://github.com/gohugoio/hugo/releases/download/v%s/hugo_%s_Linux-%s.tar.gz"
+)
+
+func downloadURL(version string) string {
+	var archType string
+	switch runtime.GOARCH {
+	case "amd64":
+		archType = "64bit"
+	case "arm64":
+		archType = "arm64"
+	case "arm":
+		archType = "arm"
+	case "386":
+		archType = "32bit"
+	default:
+		archType = "unsupported"
+	}
+	return fmt.Sprintf(_downloadURL, version, version, archType)
+}
+
+func getTempFile() (string, io.WriteCloser, error) {
+	d, err := ioutil.TempDir("", "")
```

Comment:
You should default to a different directory like the drone workspace, we already had issues in the past where tmp want writable